realm/realm-library/src/main/cpp/io_realm_internal_Collection.cpp

```diff
@@ -420,10 +350,8 @@ Java_io_realm_internal_Collection_nativeDeleteFirst(JNIEnv *env, jclass, jlong n
 
     try {
         auto wrapper = reinterpret_cast<ResultsWrapper*>(native_ptr);
-        if (wrapper->get_results().size() > 0) {
-            wrapper->get_results().get_tableview().remove(0);
-            // Refresh snapshot
-            wrapper->refresh_snapshot();
+        if (wrapper->m_results.size() > 0) {
+            wrapper->m_results.get_tableview().remove(0);
```

Comment:
@kneth By using the `move_last_over()`, `deleteFirstFromRealm` will fail because of `createNonCyclicCollection` doesn't sort the Results. But yes, we can fix it by calling the `remove()` instead. But shall we keep the original behavior to make it slower?

```
notSortedResults = {0, 1, 2, 3}
notSortedResults.deleteFromRealm(1);
// By using move_last_over it will become
notSortedResults = {0, 3, 2}
```

IMO, we should use `move_last_over()` since we never guarantee the order of unsorted `Results` is stable.